std/utf.d

```diff
@@ -3103,17 +3103,33 @@ enum dchar replacementDchar = '\uFFFD';
  * Iterate a range of char, wchar, or dchars by code unit.
  *
  * The purpose is to bypass the special case decoding that
- * $(REF front, std,array) does to character arrays.
+ * $(REF front, std,range,primitives) does to character arrays. As a result,
+ * using ranges with `byCodeUnit` can be `nothrow` while $(REF front, std,range,primitives)
+ * throws when it encounters invalid Unicode sequences.
+ *
+ * A code unit is a building block of the UTF encodings. Generally, an
+ * individual code unit does not represent what's perceived as a full
+ * character. Many characters are encoded with multiple code units. For
+ * example, the UTF-8 code units for `ø` are `0xC3 0xB8`. That
+ * means, an individual element of `byCodeUnit` often does not form a character on
+ * its own. Attempting to treat it as one while iterating over the resulting
+ * range will give nonsensical results.
+ *
  * Params:
- *      r = input range of characters, or array of characters
+ *      r = an input range of characters, or an array of characters
  * Returns:
- *      input range
+ *      An input range with a length if $(XREF traits, isAggregateType)
+ *      is `true` for `R`. Otherwise, this returns a finite random access range
+ *      with slicing.
```

Comment:
I think this description misses the other overload. The `isAggregateType` thing only matters when `isAutodecodableString!R` is true, i.e. if `R` is a `C[]` or if implicitly converts to a one (via alias this).

When `isAutodecodableString!R` is false, `byCodeUnit` returns the range untouched. I.e., in that case there are no guarantees about length, finiteness, random access, or slicing.